src/ripple/rpc/handlers/LedgerRequest.cpp

```diff
@@ -76,10 +76,19 @@ Json::Value doLedgerRequest (RPC::Context& context)
             {
                 // We don't have the ledger we need to figure out which ledger
                 // they want. Try to get it.
-                Json::Value jvResult
-                        =  getApp().getInboundLedgers().findCreate (
-                            refHash, refIndex, InboundLedger::fcGENERIC)
-                        ->getJson (0);
+
+                auto il = getApp ().getInboundLedgers ().findCreate (
+                    refHash, refIndex, InboundLedger::fcGENERIC);
+
+                if (!il)
+                {
+                    // Is rpcINTERNAL really the best error type for this?
```

Comment:
Still not sure over this.  If you look at the other code path, you're returning a `ledgerNotFound` error.  perhaps you should return that?  I'm having trouble understanding why we look up a ledger - only to return an error on success?

I think perhaps you should create an empty `jvResult`, then optionally populate it if `findCreate()` works, then inject the error as before?